garcon/activity.py

```diff
@@ -253,7 +253,16 @@ def run(self):
         previous activity is consumed (context).
         """
 
-        activity_task = self.poll()
+        try:
+            activity_task = self.poll()
+        except Exception as error:
```

Comment:
Sounds good. Thanks for the explanation. Can you please add a note around this that's a bit more descriptive of what you mentioned?